ouroboros-consensus-cardano/src/unstable-shelley-testlib/Test/Consensus/Shelley/Generators.hs

```diff
@@ -63,20 +64,20 @@ import           Test.Util.Serialisation.SomeResult (SomeResult (..))
 
 -- | The upstream 'Arbitrary' instance for Shelley blocks does not generate
 -- coherent blocks, so neither does this.
-instance (CanMock (TPraos crypto) era, crypto ~ EraCrypto era)
+instance (CanMock (TPraos crypto) era{-, Praos.PraosCrypto crypto, crypto ~ MockCrypto-})
   => Arbitrary (ShelleyBlock (TPraos crypto) era) where
   arbitrary = do
     let allPoolKeys = map snd (coreNodeKeys defaultConstants)
```

Comment:
The reason why you are running into an issue here is because we made `coreNodeKeys` monomorphic in crypto. Here is a PR that will help you solve that problem: https://github.com/IntersectMBO/cardano-ledger/pull/4877

Instead of using `coreNodeKeys` switch to:
```suggestion
    allPoolKeys <- genIssuerKeys defaultConstants
```
Same should be applied in other places where `coreNodeKeys` was used